nextclade/Snakefile

```diff
@@ -0,0 +1,375 @@
+import datetime
+
+
+wildcard_constraints:
+    flu_type="[A-Za-z0-9]+",
+    year="\d\d\d\d",
+    lineage=r"h3n2|h1n1pdm|vic|yam",
+    segment = r'pb2|pb1|pa|ha|np|na|mp|ns',
+    reference="[^_/]+",
+
+def all_builds(w):
+    builds = []
+    for lineage in config["builds"]:
+        for segment in config["builds"][lineage]:
+            for ref in config["builds"][lineage][segment]["refs"]:
+                builds.append(f"datasets/{lineage}/{segment}/{ref}/tree.json")
+    return builds
+
+rule all:
+    input:
+        all_builds
+
+
+rule download_sequences:
+    output:
+        sequences="data/{lineage}/raw_{segment}.fasta"
+    params:
+        s3_path="s3://nextstrain-data-private/files/workflows/seasonal-flu/{lineage}/{segment}/raw_sequences.fasta.xz"
+    shell:
+        """
+        aws s3 cp {params.s3_path} - | xz -c -d > {output.sequences}
+        """
+
+rule download_clades:
+    message:
+        "Downloading clade definitions for {wildcards.lineage} from {params.source_tsv} -> {output}"
+    output:
+        clade_tsv = "data/{clade}_{lineage}_{segment}_{reference}_raw.tsv"
+    params:
+        source_tsv=lambda w: config['clade_repo'] + config["builds"][w.lineage][w.segment]["clade_systems"][w.clade]['url'],
+    shell:
+        """
+        curl {params.source_tsv} > {output.clade_tsv}
+        """
+
+rule offset_clades:
+    input:
+        rules.download_clades.output,
+    output:
+        "data/{clade}_{lineage}_{segment}_{reference}.tsv",
+    params:
+        offset=lambda w: config["builds"][w.lineage][w.segment]['refs'][w.reference][
+            "clade_offset"
+        ],
+    shell:
+        """
+        perl -F'\\t' -ne \
+            '$F[2]+={params.offset} if $F[1] =~ "nuc"; \
+            print join "\\t", @F' \
+            {input} \
+            >{output}
+        """
+
+rule parse:
+    input:
+        sequences="data/{lineage}/raw_{segment}.fasta",
+    output:
+        metadata="data/{lineage}_{segment}_metadata.tsv",
+        sequences="data/{lineage}_{segment}_sequences.fasta",
+    params:
+        fields="strainName virus segment EPI_ISL date submission_date region country division location passage collecting_lab submitting_lab age sex",
+    shell:
+        """
+        augur parse \
+            --sequences {input.sequences} \
+            --fields {params.fields} \
+            --output-metadata {output.metadata} \
+            --output-sequences {output.sequences}
+        """
+
+
+def genes(w):
+    if w.segment=='ha': return ["SigPep", "HA1", "HA2"]
+    if w.segment=='na': return ["NA"]
```

Comment:
I noticed the `segment` wildcard constraints at the top of the file included other genes. Do you plan to support those other genes in Nextclade, too? It could be pretty useful for people. If so, maybe we should move this `genes` function to the top of the file next to the wildcard constraints and add the other segments to this function?

Reply:
yes, that is the plan. I moved it up and refactored it.